Hymns/Display/VerseViewModel.swift

```diff
@@ -1,28 +1,59 @@
+import Combine
 import Foundation
+import Resolver
+import SwiftUI
+
+class VerseLineViewModel: Hashable, ObservableObject {
+
+    @Published var fontSize: FontSize
 
-struct VerseLineViewModel: Hashable {
     let verseNumber: String?
     let verseText: String
     let transliteration: String?
+
+    private var disposables = Set<AnyCancellable>()
+
+    init(verseNumber: String?, verseText: String, transliteration: String?,
+         userDefaultsManager: UserDefaultsManager = Resolver.resolve()) {
+        self.verseNumber = verseNumber
+        self.verseText = verseText
+        self.transliteration = transliteration
+        self.fontSize = userDefaultsManager.fontSize
+        userDefaultsManager
+            .fontSizeSubject
+            .sink { fontSize in
+                withAnimation(.easeInOut(duration: 0.5)) {
+                    self.fontSize = fontSize
+                }
+        }.store(in: &disposables)
+    }
+
+    static func == (lhs: VerseLineViewModel, rhs: VerseLineViewModel) -> Bool {
+        lhs.verseNumber == rhs.verseNumber && lhs.verseText == rhs.verseText && lhs.transliteration == rhs.transliteration
+    }
+
+    func hash(into hasher: inout Hasher) {
```

Comment:
What does this do? How does the hash function work?

Reply:
Are you asking how a hash function works in general? https://en.wikipedia.org/wiki/Hash_function